openerp/addons/base/res/res_partner.py

```diff
@@ -500,6 +514,25 @@ def create(self, vals):
         return partner
 
     @api.multi
+    def create_company(self):
+        self.ensure_one()
+        if self.display_company_name:
+            # move info from partner to company
+            values = dict(name=self.display_company_name, is_company=True)
+            values.update(self._update_fields_values(self._address_fields()))
+            myself = dict.fromkeys(values, False) # copy before the crate which will alter the dict values
+            new_company = self.create(values)
```

Comment:
self.write({'parent_id': new_company.id,
 +                'child_ids': [(1, _id, dict(parent_id=new_company.id)) for _id in self.child_ids.ids]

})